terraso_backend/apps/soil_id/graphql/soil_id/resolvers.py

```diff
@@ -0,0 +1,272 @@
+    elif rock_fragment_volume <= 35:
+        return DepthDependentSoilData.RockFragmentVolume.VOLUME_15_35.value
+    elif rock_fragment_volume <= 60:
+        return DepthDependentSoilData.RockFragmentVolume.VOLUME_35_60.value
+    else:
+        return DepthDependentSoilData.RockFragmentVolume.VOLUME_60.value
+
+
+def resolve_soil_data(soil_match) -> SoilIdSoilData:
+    bottom_depths = soil_match["bottom_depth"]
+    prev_depth = 0
+    depth_dependent_data = [None] * len(bottom_depths)
+
+    for id, bottom_depth in bottom_depths.items():
+        depth_dependent_data[int(id)] = SoilIdDepthDependentData(
+            depth_interval=DepthInterval(start=prev_depth, end=bottom_depth),
+            texture=resolve_texture(soil_match["texture"][id]),
+            rock_fragment_volume=resolve_rock_fragment_volume(soil_match["rock_fragments"][id]),
+            munsell_color_string=soil_match["munsell"][id],
+        )
+        prev_depth = bottom_depth
+
+    return SoilIdSoilData(
+        slope=soil_match["site"]["siteData"]["slope"], depth_dependent_data=depth_dependent_data
+    )
+
+
+def resolve_ecological_site(ecological_site: dict):
+    if ecological_site["ecoclassid"] == "" or ecological_site["ecoclassid"][0] == "":
+        return None
+    else:
+        return EcologicalSite(
+            name=ecological_site["ecoclassname"][0],
+            id=ecological_site["ecoclassid"][0],
+            url=ecological_site["esd_url"][0],
+        )
+
+
+def resolve_soil_info(soil_match: dict):
+    soil_id = soil_match["id"]
+    site_data = soil_match["site"]["siteData"]
+
+    return SoilInfo(
+        soil_series=SoilSeries(
+            name=soil_id["component"],
+            taxonomy_subgroup=site_data["taxsubgrp"],
+            description=soil_match["site"]["siteDescription"],
+            full_description_url=site_data["sdeURL"],
+        ),
+        land_capability_class=LandCapabilityClass(
+            capability_class=site_data["nirrcapcl"],
+            sub_class=site_data["nirrcapscl"],
+        ),
+        ecological_site=resolve_ecological_site(soil_match["esd"]["ESD"]),
+        soil_data=resolve_soil_data(soil_match),
+    )
+
+
+def resolve_soil_match_info(score: float, rank: str):
+    return SoilMatchInfo(score=score, rank=int(rank) - 1)
+
+
+def resolve_location_based_soil_match(soil_match: dict):
+    soil_id = soil_match["id"]
+    site_data = soil_match["site"]["siteData"]
+
+    return LocationBasedSoilMatch(
+        data_source=site_data["dataSource"],
+        distance_to_nearest_map_unit_m=site_data["minCompDistance"],
+        match=resolve_soil_match_info(soil_id["score_loc"], soil_id["rank_loc"]),
+        soil_info=resolve_soil_info(soil_match),
+    )
+
+
+def resolve_location_matches_from_soil_id_result(soil_list_json: dict):
+    matches = []
+    for match in soil_list_json["soilList"]:
+        if match["id"]["rank_loc"] != "Not Displayed":
```

Comment:
question: how many special values like `Not Displayed` are there? Should they be in constants?

Reply:
maybe, but i already made a big push on code quality in this PR with the reorganization + unit testing that i wasn't sure if there was time for so i'm hesitant to keep trying to make this code really nice. i should file a tech debt issue to do this kind of cleanup though